docker/Dockerfile

```diff
@@ -24,11 +22,33 @@ ENV NODE_OPTIONS '--max-old-space-size=32768'
 # Use this for faster iterations on nginx config
 # RUN mkdir -p /app/dist  && echo hello > /app/dist/index.html
 
+# Copying files more specifically lets us reuse the build cache and iterate more quickly
+# COPY . .
+# RUN corepack enable \
+#   && yarn install \
+#   && yarn format:check \
+#   && yarn lint:check \
+#   # && yarn test \
+#   && yarn build
+
+RUN mkdir -p public src scripts .husky .yarn .storybook cypress \
```

Comment:
I can revert this, @gigxz if you want, but it was rebuilding if you touched any file at all. It's messier, but caches much better.

Reply:
I think there is a better way to do this which is just keeping `COPY . .` but adding a `.gitignore` with node_modules in it. It should make it a lot faster.  Can you try that and see if its good enough for your work setup? (And remove the `yarn cache clean`).

I worry about listing each individual (root-level) file in here, it seems like it will be hard to maintain.

Reply:
TL;DR Sure.
I'll do that, though it reruns yarn install even if your dependencies don't change. Now that I'm done iterating, though, it doesn't matter to me much. I thought it would be just the yarn.lock and package.json file, and then I kept adding things in until it worked (I clearly don't know this ecosystem very well). I think we can have a build cache for node_modules which will solve the problem more cleanly. I'll give up if it takes too long.

Reply:
I'm glad you brought this up. Definitely agree it wasn't maintainable, and what I just pushed I think is a good compromise.

Reply:
Thanks, I think this is much cleaner! Yeah the yarn install will still run, which is bothersome, but it should be relatively quick if all the node_modules are already present.
Thanks for iterating!